psi4/src/psi4/libfock/CompositeJK.cc

```diff
@@ -358,45 +447,65 @@ void DFJCOSK::compute_JK() {
         do_incfock_iter_ = (Dnorm >= incfock_conv) && !initial_iteration_ && (incfock_count_ % reset != reset - 1);
 
         if (!initial_iteration_ && (Dnorm >= incfock_conv)) incfock_count_ += 1;
-	
-	incfock_setup();
+
+        incfock_setup();
         
-	timer_off("DFJCOSK: INCFOCK Preprocessing");
+	timer_off("CompositeJK: INCFOCK Preprocessing");
     } else {
-	    D_ref_ = D_ao_;
+        D_ref_ = D_ao_;
         zero();
     }
- 
-    // Direct DF-J
+
+    // update ERI engine density matrices for density screening
+    if (density_screening_) {
+        for (auto eri_computer : eri_computers_["4-Center"]) {
+            eri_computer->update_density(D_ref_);
+	}
+    }
+
+    // => Perform matrix calculations <= //
+
+    // Coulomb Matrix
     if (do_J_) {
-        timer_on("DFJ");
-   
-   	build_J(D_ref_, J_ao_);
-   
-   	timer_off("DFJ");
+        timer_on("CompositeJK: J");
+
+        // Direct DF-J
+        if (j_type_ == "DIRECTDFJ") {
+	    build_DirectDFJ(D_ref_, J_ao_);
+	}
+
+        timer_off("CompositeJK: J");
     }
-    
-    // COSX
+
+    // Exchange Matrix
     if (do_K_) {
-        timer_on("COSK");
-   	    build_K(D_ref_, K_ao_);
-   	    timer_off("COSK");
+        timer_on("CompositeJK: K");
+
+        // LinK
+        if (k_type_ == "LINK") {
+	    build_linK(D_ref_, K_ao_);
+        // COSX
+        } else if (k_type_ == "COSX") {
+	    build_COSK(D_ref_, K_ao_);
+        }
+
+	timer_off("CompositeJK: K");
     }
 
     // => Finalize Incremental Fock if required <= //
 
     if (incfock_) {
-        timer_on("DFJCOSK: INCFOCK Postprocessing");
+        timer_on("CompositeJK: INCFOCK Postprocessing");
         incfock_postiter();
-        timer_off("DFJCOSK: INCFOCK Postprocessing");
+        timer_off("CompositeJK: INCFOCK Postprocessing");
     }
-    
+
     if (initial_iteration_) initial_iteration_ = false;
 }
 
-void DFJCOSK::postiterations() {}
+void CompositeJK::postiterations() {}
 
-void DFJCOSK::build_J(std::vector<std::shared_ptr<Matrix>>& D, std::vector<std::shared_ptr<Matrix>>& J) {
+void CompositeJK::build_DirectDFJ(std::vector<std::shared_ptr<Matrix>>& D, std::vector<std::shared_ptr<Matrix>>& J) {
```

Comment:
The changes here really only fully standardize Direct DF-J between LinK and COSX, because DFJCOSK and DFJLinK had _slight_ differences in implementation before this. Nothing here that's too important to look at unless you want to, since all of the changes here were in DFJLinK already.